keel-veto/src/main/kotlin/com/netflix/spinnaker/keel/veto/application/ApplicationVeto.kt

```diff
@@ -34,6 +34,8 @@ class ApplicationVeto(
   private val log by lazy { LoggerFactory.getLogger(javaClass) }
 
   override fun check(id: ResourceId): VetoResponse {
+    // FIXME (lpollo): The following expects a ResourceId to have a specific format, but it's not defined/enforced
+    //  anywhere. We should add a validation and perhaps token extraction methods to ResourceId to be used consistently.
     val appName = id.toString().split(":").last().split("-").first()
```

Comment:
Any resource should have the application name in its metadata. Seems like it would make sense to get it from there.

Reply:
So would you suggest to change this method to take a `Resource` instead of a `ResourceId`?

Reply:
Either that or it will need to load the full resource in here. Not sure which is more appropriate.